technologies/job/aws-lambda/functions/instanceActions.js

```diff
@@ -0,0 +1,184 @@
+          return ({
+              uuid: value,
+              status: data.State
+            })
+          }
+    ));
+    statusList=await Promise.all(statusList);
+
+    // Computing worst case status if a lambda got different triggers/eventsource
+    const xrefStatus=function(status) {
+      switch (status) {
+        case 'Creating':
+          return 30
+        case 'Enabling':
+          return 10
+        case 'Enabled':
+          return 0
+        case 'Disabling':
+          return 40
+        case 'Disabled':
+          return 50
+        case 'Updating':
+          return 20;
+        case 'Deleting':
+          return 60; 
+      }
+    }
+    const status=statusList.reduce((consolidated, item) => {
+      if (xrefStatus(item.status) > xrefStatus(consolidated))
+        return item.status;
+      else
+        return consolidated;
+    },"Enabled");
+
+    // Existing status : Creating, Enabling, Enabled, Disabling, Disabled, Updating, or Deleting
+    switch (status) {
+      case 'Creating':
+        return Response.success(JobStatus.REQUESTED);
+      case 'Enabling':
+        return Response.success(JobStatus.QUEUED);
+      case 'Enabled':
+        return Response.success(JobStatus.RUNNING);
+      case 'Disabling':
+          return Response.success(JobStatus.KILLING);
+      case 'Disabled':
+        return Response.success(JobStatus.KILLED);
+     default:
+       return Response.success(JobStatus.AWAITING); // Updating or Deleting
+    }
+  } catch (error) {
+    console.log(error);
+    return Response.error(`Failed to get status for functions ${job.featuresValues.functions.id}`, { error });
+  }
+};
+
+/**
+ * Logic to retrieve the external job instance logs.
+ * @param {Object} params
+ * @param {Object} params.job - Contains job data including featuresValues.
+ * @param {Object} params.instance - Contains instance data including the payload returned in the start function.
+ */
+exports.getLogs = async ({ job, instance }) => {
+  try {
+    console.log('GET LOG INSTANCE:', instance);
+    AWS.config.update({credentials: { accessKeyId : job.featuresValues.endpoint.aws_access_key_id, secretAccessKey:  job.featuresValues.endpoint.aws_secret_access_key}});
+    AWS.config.update({region: job.featuresValues.endpoint.region});
+
+    const cwl = new AWS.CloudWatchLogs({apiVersion: '2014-03-28'});
+
+    const paramslogstreams = {
+      logGroupName: `/aws/lambda/${job.featuresValues.functions.label}`,
+    };
+
+    const logstreams = await cwl.describeLogStreams(paramslogstreams).promise();
+
+    var logs=logstreams.logStreams.map(
+      (ls) => 
+      cwl.getLogEvents(
+        {
+          logGroupName: `/aws/lambda/${job.featuresValues.functions.label}`,
+          logStreamName: ls.logStreamName
+        }
+        ).promise().then((data) => {
+          return (
+              data.events
+            )
+          }
+    ));
+    
+    logs=(await Promise.all(logs)).flat();
```

Comment:
Just to be sure, and I don't know the return of the CloudWatchLogs API, this flatten only one level of array like:

```js
const arr1 = [1, 2, [3, 4, [5, 6]]];
arr1.flat();
// [1, 2, 3, 4, [5, 6]]
```

But if you want, you can give `Infinity` parameter to the `flat()` function just to make sure that everything is on the same level.

```js
const arr2 = [1, 2, [3, 4, [5, 6, [7, 8, [9, 10]]]]];
arr2.flat(Infinity);
// [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
```

Reply:
it's only an array of arrays of events. So flat() is good.

Reply:
:+1: